.evergreen.yml

```diff
@@ -46,87 +90,11 @@ functions:
-            cd \$CHANGE_DIR_TO;
-            echo "Changed cwd on cygwin. Current working dir: \$(pwd)"
-          fi
-
-          # Make default evergreen expansions accessible to hadron-build
-          export EVERGREEN_ASSET_PREFIX="s3://mciuploads/${project}/${revision}";
-          export EVERGREEN_AUTHOR="${author}";
-          export EVERGREEN_AWS_ACCESS_KEY_ID="${aws_key}";
-          export EVERGREEN_AWS_SECRET_ACCESS_KEY="${aws_secret}";
-          export EVERGREEN_BRANCH_NAME="${branch_name}";
-          export EVERGREEN_BUILD_ID="${build_id}";
-          export EVERGREEN_BUILD_VARIANT="${build_variant}";
-          export EVERGREEN_EXECUTION="${execution}";
-          export EVERGREEN_IS_PATCH=${is_patch};
-          export EVERGREEN_PROJECT="${project}";
-          export EVERGREEN_REVISION="${revision}";
-          export EVERGREEN_TASK_ID="${task_id}";
-          export EVERGREEN_TASK_NAME="${task_name}";
-          export EVERGREEN_TASK_URL="https://evergreen.mongodb.com/task/${task_id}";
-          export EVERGREEN_VERSION_ID="${version_id}";
-          export EVERGREEN_WORKDIR="${workdir}";
-
-          export HADRON_METRICS_BUGSNAG_KEY="${metrics_bugsnag_key}";
-          export HADRON_METRICS_INTERCOM_APP_ID="${metrics_intercom_app_id}";
-          export HADRON_METRICS_STITCH_APP_ID="${metrics_stitch_app_id}";
-          export HADRON_METRICS_SEGMENT_API_KEY="${metrics_segment_api_key}";
-
-          # Directories used to install node, npm, and store artifacts
-          export ARTIFACTS_PATH="$(pwd)/.deps"
-          export NPM_CACHE_DIR="$(pwd)/.deps/.npm"
-          export NPM_TMP_DIR="$(pwd)/.deps/tmp"
-
-          export NODE_JS_VERSION="14.17.5"
-          export NPM_VERSION="7"
-
-          # npm configuration
-          # all var names need to be lowercase
-          # see: https://docs.npmjs.com/cli/v7/using-npm/config#environment-variables
-          export npm_config_cache="\$NPM_CACHE_DIR"
-          # npm tmp is deprecated, but let's keep it around just in case
-          export npm_config_tmp="\$NPM_TMP_DIR"
-
-          if [[ "\$OSTYPE" == "cygwin" ]]; then
-            # NOTE lucas: for git-core addition, See
-            # https://jira.mongodb.org/browse/COMPASS-4122
-            export PATH="/cygdrive/c/Program Files/Git/mingw32/libexec/git-core/:$(pwd)/.deps:/cygdrive/c/wixtools/bin/:\$PATH"
-            export APPDATA=Z:\\\;
-          fi
-
-          if [[ "\$OSTYPE" != "cygwin" ]]; then
-            export PATH="$(pwd)/.deps/bin:\$PATH"
-          fi
-
-          if [ "\$EVERGREEN_BUILD_VARIANT" = rhel ]; then
-            # To build node modules on RHEL post electron 13 we need
-            # a newer c++ compiler version, this adds it.
-            # https://jira.mongodb.org/browse/COMPASS-5150
-            export PATH="/opt/mongodbtoolchain/v3/bin:\$PATH"
-          fi
-
-          EOF_BUILD_SH
-
-          source ~/compass_env.sh
+          set -e
+          eval $(.evergreen/print-compass-env.sh)
```

Comment:
This is the bit that solves the whole main issue: The expansions that don't need further processing are mixed in as env vars and the ones that do get output as a bash script to be eval'd in here.

The contents of this `script:` block will end up in logs. Previously we did the expansions in the code so it essentially becomes a "hardcoded" block of exports - ie the script block contains the exports with the secrets in plain text. Whereas env vars or the output of .evergreen/print-compass-env.sh don't end up visible in here but only  the few remaining lines of shell script containing no secrets themselves.